contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -706,23 +706,21 @@ GRANT ALL on PROCEDURE sys.sp_describe_first_result_set TO PUBLIC;
 
 CREATE OR REPLACE VIEW sys.spt_tablecollations_view AS
     SELECT
-        o.object_id         AS object_id,
-        o.schema_id         AS schema_id,
-        c.column_id         AS colid,
-        CASE WHEN p.attoptions[1] LIKE 'bbf_original_name=%' THEN CAST(split_part(p.attoptions[1], '=', 2) AS sys.VARCHAR)
-			ELSE c.name COLLATE sys.database_default END AS name,
+        c.object_id                      AS object_id,
+        CAST(p.relnamespace AS int)      AS schema_id,
+        c.column_id                      AS colid,
+        CAST(c.name AS sys.varchar)      AS name,
```

Comment:
I would suggest you to replace all the views with pg_catalog. For example, c.object_id is same as pg_class.oid. You can check how are we obtaining the value of a particular column from the underlying view definition. IMO, if we can get rid of using all view and use only pg catalogs, that would increase the performance better.

NOTE: Make sure that we are obtaining the set of results back.